pp_ctl.c

```diff
@@ -3485,7 +3485,7 @@ S_doeval_compile(pTHX_ U8 gimme, CV* outside, U32 seq, HV *hh)
 
     SAVEHINTS();
     if (clear_hints) {
-	PL_hints = 0;
+	PL_hints = HINT_DEFAULT;
```

Comment:
@tonycoz I am 99% certain this is the correct thing to do here but admit to not being clear under what the current circumstances are that we want to set it to 0?

Reply:
They're set to 0 for require, so this change looks right.